system/temperature.c

```diff
@@ -47,14 +182,11 @@ int get_cpu_temperature(void)
     }
 
     // AMD CPU
+    if (cpuid_info.vendor_id.str[0] == 'A' && cpuid_info.version.extendedFamily == 0) {
+        return k8_temp();
+    }
     if (cpuid_info.vendor_id.str[0] == 'A' && cpuid_info.version.extendedFamily > 0 && cpuid_info.version.extendedFamily < 8) {
-
-        // Untested yet
-        uint32_t rtcr = pci_config_read32(0, 24, 3, 0xA4);
-        int raw_temp = (rtcr >> 21) & 0x7FF;
-
-        return raw_temp / 8;
```

Comment:
Is there anything wrong with this simple code (expect errata 319 handling)?
It basically do the same thing (read 0/24/3/0xA4, shift 21, divide by 8) as all code added before.

Reply:
0/24/3 is an internal register from the BSP. It can't be absent, hidden or anywhere else on any CPU targeted by the previous case (`cpuid_info.vendor_id.str[0] == 'A' && cpuid_info.version.extendedFamily > 0 && cpuid_info.version.extendedFamily < 8`).

Are you aware of any real-word platform where 0/24/3 does not exist?

Reply:
From the few hardware I have, no.
I don't see any reason to not add a little test for being safe but if you say you know for sure it will exists at that place, I will dont do any objections.